File: learn_android/code/ParcelableDemo1/app/src/main/java/com/hkc/parcelabledemo1/MainActivity.java
package com.hkc.parcelabledemo1;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.hkc.parcelabledemo1.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
        binding = DataBindingUtil.setContentView(this,R.layout.activity_main);

        binding.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String name = binding.editTextName.getText().toString();
                int age = Integer.valueOf(binding.editTextAge.getText().toString());
                int Math =  Integer.valueOf(binding.editTextMath.getText().toString());
                int English = Integer.valueOf(binding.editTextEnglish.getText().toString());
                Student student = new Student(name,age,new Score(Math,English));

                MyApplication myApplication = (MyApplication) getApplication();
                myApplication.student = student;

                Intent intent = new Intent(MainActivity.this,MainActivity2.class);
                //直接放入
//                intent.putExtra("student",student);
                //先打包成Bundle
//                Bundle bundle = new Bundle();
//                bundle.putParcelable("student",student);
//                intent.putExtra("data",bundle);
                startActivity(intent);

            }
        });




    }
}

File: learn_android/code/ServiceTest/app/src/main/java/com/hkc/servicetest/MainActivity.kt
package com.hkc.servicetest

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //為按鈕設定監聽器
        findViewById<Button>(R.id.btn_start).setOnClickListener {
            //使用startService()方法啟動Service
            startService(Intent(this,MyService::class.java))

            Toast.makeText(this,"啟動Service", Toast.LENGTH_LONG)
            //關閉Activity
            finish()

        }

    }
}